examples/vision/siamesenetwork.py

```diff
@@ -0,0 +1,417 @@
+"""
+Title: Siamese Network with Triplet Loss
```

Comment:
Suggestion: add a bit more info about the task to help the user understand why they'd want to check this example

```suggestion
Title: Siamese Network to compare similarity with a triplet loss
```

or something like that